### ISSUE TITLE
Fix ResizableControl::DisableResizing

### ISSUE THREAD

**Issue by ipo** [billyquith/GWork#41]
I could not get DisableResizing to work properly for the current checkout I have of the gwork-branch.  Perhaps I was doing something wrong, I am just beginning to use the lib so I'm not entirely familiar with the ins and outs.  My opengl-renderer also doesn't implement scissoring yet so that could possibly be the cause, but from how the issue appears I don't think so.

I wrote a patch to fix it.  You can have a look at http://i.imgur.com/65sJ5fe.png to see the issue I was having, one column with WindowControl with resizing enabled, and another with it disabled.  The rows shows how it looks pre- and post-patch.

Since I can't create a pull-request without having done all the repo-work on github, all I can offer is this diff of my changes: https://gitlab.com/jonas.lindgren/gwork/commit/3555e6000ca5152fb1df7df5dd065747a5f9c8a3.diff

The code snippet necessary to showcase the problem goes something like this:
>Gwk::Controls::WindowControl* window = new Gwk::Controls::WindowControl(canvas);
>window->SetTitle("Space survival - Start menu");
>window->SetSize(300, 300);
>window->SetPos(
>  1280/2 - window->GetSize().x/2,
>  720/2 - window->GetSize().y/2
>  );
>window->SetDeleteOnClose(true);
>window->DisableResizing();

**Comment by billyquith** [comment]
To make a pull request here all you have to do is fork Gwork and then push your change to that repo. It's just another remote.

**Comment by ipo** [comment]
If I were actively using github that would be very convenient for everyone involved, but as it is now I'm not interested in jumping through those hoops.  It would be nice for my bugfixes to be made available to others, to improve the lib for everyone, but ultimately I have nothing personally to gain from it.

Applying a diff is trivial, through [git apply](https://git-scm.com/docs/git-apply) , far easier than for me to cajole github into accepting the pull-request.  If anyone is interested in the fix and you insist that someone create a pull-request, I welcome them to create one.  If not we can instead consider this a bug report.

**Comment by billyquith** [comment]
That's a shame. I was hoping to build a community of regular contributors. I've spent a lot of my personal time refactoring this library so others could benefit.

**Comment by ipo** [comment]
I agree.  I was surprised that github had no way of making a pull-request other than within the very rigidly defined framework of 'fork on our site then generate a pull-request only from that repository', but the cynic in me thinks this makes sense from a community-building standpoint.  No point helping the competition, as it were.

**Comment by eXpl0it3r** [comment]
It would've taken you less time to actual do it, than to write the above text. But hey, principles need to be followed...

PR at #42

**Comment by ipo** [comment]
This will be a recurring argument whenever I patch anything, so I guess I have to use github after all.

**Comment by billyquith** [comment]
>  I was surprised that github had no way of making a pull-request other than within the very rigidly defined framework of 'fork on our site then generate a pull-request only from that repository', 

I think this is to simplify things and avoid outside problems. What if the external site disappears or changes? Are all external sites configured the same? How can you monitor for notifications, etc. It is easier and less error prone to just push your changes to the clone on Github. 

There is nothing to stop you developing in git repo anywhere and then just pushing the changes you want to contribute to your fork on Github. The thing about git is that there are multiple copies of a project repo and any of them can arbitrarily be the main one. All the others are just relative to the source (by parent lineage). It is a more flexible way of working than old client-server model, e.g. like Subversion.

You might even have multiple forks yourself, perhaps you included Gwork as a sub-repo in a larger project, so you might have multiple clones on the same machine. When you make a change you want to contribute you just push them to the remote. Since I own the original remote you need to show me a fork with your changes in so that a PR can be created and the changes discussed and tested before merging. And you could push from any of your forks to your reference version on Github.

I'll just add that having a workflow simplifies this, e.g. creating feature/bugfix branches, before changing and creating PR. This stops multiple changes being conflated. Not all your changes might make it into the original repo, for various reasons.

**Comment by billyquith** [comment]
Perhaps I should try and answer your original question first.

> I could not get DisableResizing to work properly for the current checkout I have of the gwork-branch. Perhaps I was doing something wrong, I am just beginning to use the lib so I'm not entirely familiar with the ins and outs.

TBH I am still learning how much of it works! There is very little documentation or useful commenting in the original code so sometimes you have to make assumptions about how things work. I'll try and add online docs at some point.

> My opengl-renderer also doesn't implement scissoring yet so that could possibly be the cause, but from how the issue appears I don't think so.

There is an OpenGL sample and renderer in Gwork.  You could try using that renderer with this code to see if you get the same problem, if you haven't yet implemented all the features. Perhaps just paste the code into the sample? Try this first.

**Comment by ipo** [comment]
I updated the test-executable to make it possible to easily test DisableResizing() with it.  In the process I confirmed that the error exists with the built-in OpenGL-renderer.  Clipping also seems to be nonfunctional in that one as well.

I have updated the pull request to include the test.

**Comment by billyquith** [comment]
> Clipping also seems to be nonfunctional in that one as well.

I just looked at the OpenGL sample and clipping looks fine. E.g. tree control, collapsable list, etc. What problem are you seeing?

**Comment by billyquith** [comment]
So I looked into this and there was a bug with the resizer modification, as you noticed. Should be fixed now.

I also added a test for non-resizable window in the demo.

**Comment by ipo** [comment]
Wow, you just so happened to write a fix that was more or less the same as the pull-request, instead of using the pull-request.  That's vaguely insulting.

**Comment by billyquith** [comment]
I thought the problem was elsewhere and then ended up with a similar solution. I also tidied up some other stuff. Apologies, but there were issues with the PR that needed correcting before submission:

- `m_disabled` is duplicate. See `m_bDisabled`. Also access public.
- Extra unnecessary line spaces.
- Redundant code left commented in.
- No test in the demo.

The etiquette is to fix these in the PR before submission. I also looked at GWEN for similar issues and found a similar solution.

**Comment by billyquith** [comment]
Ok, this is fixed now I believe. I'll close this issue.

You mentioned some problem with clipping the OpenGL sample. I can't see evidence of this. If you still find a problem please open a new issue.

I'm happy to accept PRs but not ones which I end up fixing/rewriting. Other PRs have made it in. Overall I'm trying to improve the code quality here (see above points).

**Comment by ipo** [comment]
>No test in the demo.

What on earth are you talking about?
See https://github.com/billyquith/GWork/pull/44/commits/915ae999b69d933e4a556d8bb77111a50bf5631f , committed 15 hours before "you fixed it".

The other points are fair and could easily have been adjusted.

You are missing the main point.  On one hand you are claiming that you want to build a community, but on the other hand when a good-faith attempt to collaborate comes along, you would rather replicate the work on your own and do your own commits, than actually collaborate.

It should not surprise you that this does not build
>a community of regular contributors

**Comment by eXpl0it3r** [comment]
I think this is a rather unfair stand to take, especially after your initial reluctancy to send a PR at all.

It can happen at any point in time that two people work on a solution for the same problem in parallel. It's only fair for any maintainer to prefer their own solution which they can be certain to work and don't need additional adjustments.

Besides, you didn't want to take credit for the initial patches anyways, so why are you getting work up over seeing your PR not being merged?

And finally, can I also get upset that you first refuse to utilize GitHub and once I do the "work" for you, you suddenly decide to use it after all and dismiss one of my PRs? I was not really happy about this either, yet I didn't feel the need to mention it until now...

Contributions are always welcome but one also has to accept that something might not make it in or that a very similar solution from someone else is being used in the end.

**Comment by billyquith** [comment]
Thank you. I was about to say the exact same thing.

**Comment by ipo** [comment]
>can I also get upset that you first refuse to utilize GitHub and once I do the "work" for you, you suddenly decide to use it after all and dismiss one of my PRs?

What are you referring to here?  The one that needed to be updated, which I could not with a PR you owned, leaving me with creating another as the only option?  Or the other one you created, which I have *not* replicated so as to not have two identical PRs?  Which one did you feel was unfair and rude of me?

Ultimately, I took your point that I should create a PR.  If you want to feel upset about that, I can't stop you, but I have certainly not "dismissed" your PR.  I updated it the only way I possibly could.

>yet I didn't feel the need to mention it until now...

You did reply with a passive-aggressive "..." when I pointed out I had to update, didn't you?  Let's not pretend you're the virtuous saint after the fact.

Ultimately, I cannot force @billyquith to do anything at all, and that is not my intent.  I'm just pointing out that I came here to improve the lib as best I could, but as a result of how this was handled I won't do that again.  If @billyquith is truly interest in creating a community of regular contributors, as he has stated, that's something he should be interested in knowing, for future reference.
